src/astro/databases/google/bigquery.py

```diff
@@ -135,3 +146,100 @@ def merge_table(
             )
             statement += f" WHEN MATCHED THEN {update_statement}"
         self.run_sql(sql_statement=statement)
+
+    def check_optimised_path(self, source_file: File, target_table: Table) -> bool:
+        """
+        Check if there is an optimised path for source to destination.
+
+        :param source_file: File from which we need to transfer data
+        :param target_table: Table that needs to be populated with file data
+        """
+        return bool(self.OPTIMIZED_PATHS.get(source_file.location.location_type))
+
+    def optimised_transfer(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ):
+        """
+        Checks if optimised path for transfer between File location to database exists
+        and if it does, it transfers it and returns true else false.
+        """
+        method_name = self.OPTIMIZED_PATHS.get(source_file.location.location_type)
+        if method_name:
+            transfer_method = self.__getattribute__(method_name)
+            transfer_method(
+                source_file=source_file,
+                target_table=target_table,
+                if_exists=if_exists,
+                **kwargs,
+            )
+
+    @staticmethod
+    def get_project_id_from_conn(conn: connection) -> str:
+        """
+        Get project id from conn
+
+        :param conn: Airflow's connection
+        """
+        if conn.extra and conn.extra_dejson.get("project"):
+            return str(conn.extra_dejson["project"])
+        elif conn.host:
+            return str(conn.host)
+        raise ValueError(f"conn_id {conn.conn_id} has no project id.")
+
+    def gs_to_bigquery(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ) -> None:
+        """
+        Transfer data from gcs to bigquery
+
+        :param source_file: Source file that is used as source of data
+        :param target_table: Table that will be created on the bigquery
+        :param if_exists: Overwrite table if exists. Default 'replace'
+        """
+        bq_hook = BigQueryHook(
+            gcp_conn_id=target_table.conn_id,
+        )
+
+        if if_exists == "replace":
+            self.drop_table(target_table)
```

Comment:
We may want to do this regardless if it is an optimised implementation or not. Would there be a higher level function where we could place this?